api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
+        address_params = address_book_params
+        default_flag = address_params.delete(:default)
+        address = current_api_user.save_in_address_book(address_params, default_flag)
+        if address.valid?
+          @user_addresses = current_api_user.user_addresses
+          render :mine, status: :ok
```

Comment:
I think that this is a good place to introduce `303 See Other` HTTP status. I use it in my projects, to minimize amount of duplicated code.

Currently I see that in 3 places code `@user_addresses = current_api_user.user_addresses` is used. You can move it to some method, to avoid duplications, but I prefer solution with redirect. When non-GET action is successfull, then server can respond with `redirect_to action: :mine, code: 303`. This means: Everything was successfull, to get information about current resource see this URL.

I know that this is another request, but browsers follow that redirections automatically.